protos/sigstore_common.proto

```diff
@@ -168,25 +168,17 @@ message SubjectAlternativeName {
         }
 }
 
-// A chain of X.509 certificates.
+// A collection of X.509 certificates.
+//
+// NOTE: "Chain" is a misnomer in this context, since there is no one true certificate chain
+// in most PKI schemes. This message should be treated as a generic collection of certificates
+// for path construction.
 message X509CertificateChain {
-        // The chain of certificates, with indices 0 to n.
-        // The first certificate in the array must be the leaf
-        // certificate used for signing.
-        //
-        // Signers MUST NOT include their root CA certificates in their embedded
-        // certificate chains, and SHOULD NOT include intermediate CA
-        // certificates that appear in independent roots of trust.
-        //
-        // Verifiers MUST validate the chain carefully to ensure that it chains
-        // up to a root CA certificate that they trust, regardless of whether
-        // the chain includes additional intermediate/root CA certificates.
-        // Verifiers MAY enforce additional constraints, such as requiring that
-        // all intermediate CA certificates appear in an independent root of
-        // trust.
+        // One or more DER-encoded certificates.
         //
-        // Verifiers SHOULD handle old or non-complying bundles that have
-        // additional intermediate/root CA certificates.
+        // In some contexts (such as `VerificationMaterial.certificate`), this sequence
+        // has an imposed order. Unless explicitly specified, there is otherwise no
+        // guaranteed order.
```

Comment:
It's only a comment so we can't enforce easily, but should we be more prescriptive regarding order? Something like "index n+1 SHOULD be the issuing certificate for index n"?

Reply:
We could do that, although there may be (I really hope nobody does this, but there's nothing stopping it) cross-issuing PKI topologies like this one:

```
RCA_A -> ICA_A'
RCA_B -> ICA_A''
ICA_A -> leaf
```

(i.e. `ICA_A` is a single logical CA with two separate CA certs issued by different root CAs, either of which can be a valid issuer for `leaf`).

In this topology, `N+1` could be either `ICA_A'` or `ICA_A''`, and both could be listed in the "chain" for completeness purposes. So it's not clear which to order first 🙂 

(In practice, maybe this doesn't matter since `SHOULD` isn't `MUST`?)

Reply:
I agree with @haydentherapper on being more prescriptive here. And as @woodruffw points out, if someone is doing a multi-rooted tree, they are are allowed as it's a `SHOULD`, and also, the verifier would probably be aware of this (i.e have both `RCA_A` and `RCA_B`) and should so be able to construct a chain regardless.

Reply:
Sounds good, I'll make it more prescriptive!

Reply:
Is there some situation in which we're expected to parse out this multirooted chain?

Reply:
6d2a288 strengthens the prescription in the bundle case, PTAL 🙂

Reply:
> Is there some situation in which we're expected to parse out this multirooted chain?

This could show up in a BYO PKI scenario where the root of trust contains multiple separat trust anchors, but I think it's pretty unlikely -- I mostly produced it to illustrate that the chain "order" is ill-defined outside of basic PKIs.